extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java

```diff
@@ -802,191 +808,120 @@ public boolean isCoordinator() {
-                        memberHandler.onValueReceived("CurrentArtist", (artist != null) ? artist : "", "AVTransport");
-                        memberHandler.onValueReceived("CurrentTitle", (title != null) ? title : "", "AVTransport");
-                        memberHandler.onValueReceived("CurrentAlbum", (album != null) ? album : "", "AVTransport");
-                    }
-                } catch (IllegalStateException e) {
-                    logger.warn("Cannot update track data for group member ({})", e.getMessage());
-                }
-            }
-            updateMediaInfo();
-
-        } catch (IllegalStateException e) {
-            logger.warn("Cannot update track data ({})", e.getMessage());
+        if (currentURI == null) {
+            // Do nothing
         }
-    }
-
-    protected void updateCurrentURIFormatted(String URI) {
-
-        String currentURI = URI;
-        SonosMetaData currentTrack = null;
-        SonosMetaData currentUriMetaData = null;
-
-        try {
-            ZonePlayerHandler coordinatorHandler = getCoordinatorHandler();
 
-            if (coordinatorHandler != null && coordinatorHandler != this) {
-                if (currentURI.contains(LINE_IN_URI)) {
-                    coordinatorHandler.updateMediaInfo();
-                }
-                currentURI = coordinatorHandler.getCurrentURI();
-                currentTrack = coordinatorHandler.getTrackMetadata();
-                currentUriMetaData = coordinatorHandler.getCurrentURIMetadata();
-            } else {
-                // currentURI = getCurrentURI();
-                currentTrack = getTrackMetadata();
-                currentUriMetaData = getCurrentURIMetadata();
-            }
-
-        } catch (IllegalStateException e) {
-            logger.warn("Cannot handle current URI ({})", e.getMessage());
-            currentURI = null;
-            currentTrack = null;
+        else if (currentURI.isEmpty()) {
+            // Reset data
+            needsUpdating = true;
         }
 
-        if (currentURI != null) {
-            String title = stateMap.get("CurrentTitle");
-            String resultString = stateMap.get("CurrentURIFormatted");
-            boolean needsUpdating = false;
+        else if (currentURI.contains(GROUP_URI)) {
+            // The Sonos is a slave member of a group, we do nothing
+            // The media information will be updated by the coordinator
+            // Notification of group change occurs later, so we just check the URI
+        }
 
-            if (opmlUrl != null && currentURI.contains(STREAM_URI)) {
+        else if (currentURI.contains(STREAM_URI)) {
+            // Radio stream (tune-in)
+            boolean opmlUrlSucceeded = false;
+            if (opmlUrl != null) {
                 String stationID = StringUtils.substringBetween(currentURI, ":s", "?sid");
-                String previousStationID = stateMap.get("StationID");
-
-                Calendar now = Calendar.getInstance();
```

Comment:
When you select a new radio station, you will have two notifications in a short delay, one with the state TRANSITIONING and one with the state PLAYING. With my current code, this will trigger 2 OPML requests in a short delay for the same station. To optimize and reduce a little more the OPML requests, we could decide to not update media information when state is TRANSITIONING ?